common/faceting.js

```diff
@@ -685,19 +690,87 @@
                         }
                     }
 
+                    var FLOAT_PRECISION = 10000;
+                    function initializeRangeMinMax(min, max) {
+                        if (isColumnOfType('timestamp')) {
+                            if (!min) {
+                                scope.rangeOptions.absMin = null;
+                            } else {
+                                // incase of fractional seconds, truncate for min
+                                var m = moment(min).startOf('second');
+                                scope.rangeOptions.absMin = {
+                                    date: m.format(dataFormats.date),
+                                    time: m.format(dataFormats.time24)
+                                }
+                            }
+
+                            if (!max) {
+                                scope.rangeOptions.absMax = null;
+                            } else {
+                                // incase of fractional seconds, add 1 and truncate for max
+                                var m = moment(max).add(1, 'second').startOf('second');
+                                scope.rangeOptions.absMax = {
+                                    date: m.format(dataFormats.date),
+                                    time: m.format(dataFormats.time24)
+                                }
+                            }
+                        } else if (isColumnOfType('float')) {
+                            // epsilon can be calculated using Math.pow(2, exponent_base + log2(x))
+                            // check for float cases for extra precision
+                            var minEps = 0, maxEps = 0,
+                                tiny, expbase;
+
+                            // use the max of tiny and epsilon
+                            // max(tiny, pow(2, expbase + log2(abs(x))))
+                            // log2(0) and log2(-x) are undefined so use absolute values
+                            if (isColumnOfType('float4')) {
+                                tiny = Math.pow(2, -127); // min exponent -127
+                                expbase = -23; // 23 bit mantissa
+                            } else if (isColumnOfType('float8')) {
+                                tiny = Math.pow(2, -1022); // min exponent -1022
+                                expbase = -52; // 52 bit mantissa
+                            }
+
+                            // max(tiny, pow(2, expbase + log2(abs(x))))
+                            // use tiny if abs(x) is 0 or negative
+                            minEps = (Math.abs(min) > 0) ? Math.max( tiny, Math.pow(2, expbase + Math.log2(Math.abs(min))) ) : tiny;
```

Comment:
I'm trying to understand this line, isn't the result of `Math.abs` always 0 or positive? So it will only check if min is not zero?